sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -82,94 +101,82 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
     super(nextPolicy, options);
   }
 
-  private parseWWWAuthenticate(www_authenticate: string): {
-    authorization: string,
-    resource: string
-  } {
-    const returnValue = {
-      authorization: "",
-      resource: ""
-    };
-    // Parses an authentication message like:
-    // ```
-    // Bearer authorization="some_authorization", resource="https://some.url"
-    // ```
-    let spaceSep = www_authenticate.split(" ");
-
-    // Split the KV comma-separated list
-    for (const spaceItem of spaceSep) {
-      const commaSep = spaceItem.split(",");
-      for (const commaItem of commaSep) {
-        // Split the key/value pairs
-        const kv = commaItem.split("=");
-        const key = kv[0].trim();
-        const removeQuotes = (x: string): string => x.trim().replace(/['"]+/g, '');
-        if (key == "authorization" || key == "authorization_uri") {
-          returnValue.authorization = removeQuotes(kv[1]);
-        } else if (key == "resource" || key == "scope") {
-          returnValue.resource = removeQuotes(kv[1]);
-        }
-      }
-    }
-    return returnValue;
-  }
-
   /**
    * Applies the Bearer token to the request through the Authorization header.
    * @param webResource
    */
-  public async sendRequest(
-    webResource: WebResource
-  ): Promise<HttpOperationResponse> {
-    if (!webResource.headers) webResource.headers = new HttpHeaders();
+  public async sendRequest(webResource: WebResource): Promise<HttpOperationResponse> {
+    // Headers must exist in the webResource object.
+    if (!webResource.headers) {
```

Comment:
are headers ever missing? I thought it was required?

Reply:
Checking!

Reply:
Removed!